feat:Bind Ctrl-v to visually editing the selected command, exactly as per fc

wolf opened the pull request:
It took me a while to get back to this and finally make the needed changes, but here it is at last.

My old workflow was:
  - select the command I want
  - hit tab, so now it is on the command line, atuin is gone, nothing has been executed
  - I use Vim mode on the command line, so Escape puts me in normal mode
  - vv launches my favorite editor
  - edit, write, and quit which makes the updated command execute

My new workflow follows in the footsteps of fc:
  - select the command I want
  - hit Ctrl-v, launches my favorite editor on the text of that command
  - edit, write, and quit which makes the updated command execute and is saved in history

I initially engaged on this topic roughly a year ago. I tried to make this as much like fc as I could because I feel like atuin hides and replaces fc.  Emacs-mode on the command line requires two control keys to jump into the editor.  Vim-mode needs you to get into normal mode, that is NOT inserting, then hit v twice (well ... twice in Zsh, I think just once in actual readline).  Everything else in atuin is a single key, and when stepping through the history list, you're already not in insert mode.  Like fc, I look for editors in this order:
  - the value of $FCEDIT (I know this feels weird, but you are "fixing a command")
  - the value of $VISUAL
  - the value of $EDITOR
  - /usr/bin/editor (Debian/Ubuntu update-alternatives symlink, if present)
  - vim or vi (whichever is found in PATH first)

## Implementation notes

The action is wired through the keymap system as `edit-accept`, bound to `ctrl-v` by default in both emacs and vim-normal keymaps. It can be rebound in `config.toml` like any other action.

The editor command is parsed with `shlex` so values like `VISUAL="nvim -u NORC"` work correctly.

Before spawning the editor, the TUI terminal is fully torn down (raw mode disabled, alternate screen exited). The editor's stdin/stdout/stderr are explicitly redirected to `/dev/tty` so that ncurses-based editors (nano, vim) work correctly — without this, the shell integration's fd-swap for command substitution causes arrow keys to break and terminal escape sequences to leak into the shell's captured output.

I am very open to discussion.  I want to know if this is a desirable feature, the right implementation, in the right place, with the right names.  The functionality is more important to me than exactly how I implemented it.  And atuin staying true to itself is even more important than that.  I originally posted in the CLI channel on the Discord server describing this.

## Checks
- [x] I am happy for maintainers to push small adjustments to this PR, to speed up the review cycle
- [x] I have checked that there are no existing pull requests for the same thing

wolf commented on crates/atuin/src/command/client/search/interactive.rs:
```diff
@@ -84,6 +88,36 @@ struct StyleState {
+    std::env::var("FCEDIT")
+        .or_else(|_| std::env::var("EDITOR"))
+        .or_else(|_| std::env::var("VISUAL"))
+        .unwrap_or_else(|_| "vim".to_string())
```
Using `vi` instead of `vim` sounds right.  In systems with `vim`, `vi` is usually just another name for `vim`.  A thing I was thinking, related to this, is that perhaps `config.toml` could optionally specify an editor, and that would be something to check for first.  Reporting an error seems like user-friendly behavior.  Is that the general way atuin handles situations like this?  With respect to the exact editors used, I feel like I would want to probe for `vi` and issue the error only after that.  That way I maintain the same sequence used by `fc` (modulo any change in `config.toml`).  But maybe my desire to emulate `fc` is not appropriate.

When you say `/usr/bin/editor`, you're just using `editor` as a stand-in, right?  Just double-checking.  You're asking that I make sure no matter what they have specified, even in one of those variables, that their choice actually exists?

Finally, it occurs to me that this is a feature you might want to turn on/off in `config.toml`.  It seems unlikely, at least to me, but it's a thought.  I feel like "off" means "just don't touch Ctrl-v".  Hmm.  And also I should look to see any place else that mentions keys and consequences, e.g., help text, documentation, etc.  A PR that adds a key should update such things as well.

It wasn't immediately obvious to me how to write a test (or a couple of tests) in the test section of `interactive.rs` since running a visual editor requires interactive input.  This is on my mind.

wolf replied:
Oh I see, `editor` is not just a place-holder.  On some OSs, that's an actual thing.

wolf replied:
I think I know how to do this as your are suggesting.  Making sure specific editors exist feels like the code will be bulky compared to what is currently in this PR.  Looking specifically for `/usr/bin/editor`; and maybe adding a key to `config.toml`.  But I also think these are good ideas, regardless of whether you are a maintainer or not.

I will explore implementing this, and update this PR for further review.

jeremycline replied:
Indeed, I wouldn't enumerate `nvim`, `emacs`, etc. As long as the probe for `/usr/bin/editor` happens after checking the `EDITOR` environment variable I don't think you need to worry about checking the OS, either, although that might be necessary if some distribution ships something else as `/usr/bin/editor` and also don't set an `EDITOR` environment variable.

Personally, I feel like we can get away without a `config.toml` setting for the editor since the `EDITOR` environment variable is so widely used and I have a hard time imagining someone who wants to use `nvim` for git commit messages, but `emacs` for editing commands in Atuin.

For whatever it's worth, I lean _slightly_ towards presenting the user with an error over defaulting to `vi` if nothing is set. I expect nearly every Atuin user knows how to use `vi`, but for any that don't it would be a shocking experience to accidentally end up there. I don't have a great sense of whether or not that matches the general Atuin style though.

[There's a table of shortcuts](https://docs.atuin.sh/configuration/key-binding/#atuin-ui-shortcuts), but I'm not sure where the source for that document is.

Finally, for testing maybe you could set `EDITOR` to a shell script that writes a string to the given tempfile so it's not interactive? I've not thought hard about that, though.

wolf committed:
feat: edit-accept action — open selected command in $VISUAL before executing

Adds an fc(1)-style "edit then execute" flow to the interactive search TUI.
Pressing ctrl-v opens the currently selected history entry in the user's
preferred editor; on save and exit the edited command executes and is
recorded in Atuin history, matching POSIX fc behaviour.

Editor resolution order: $FCEDIT → $VISUAL → $EDITOR → /usr/bin/editor
(Debian/Ubuntu update-alternatives) → vim → vi (both verified present
before use).

Implementation notes:
- New Action::EditAccept ("edit-accept") wired through the keymap system;
  bound to ctrl-v in emacs and vim-normal default keymaps, user-rebindable
- Editor command parsed with shlex so values like VISUAL="nvim -u NORC" work
- Terminal torn down (drop(terminal)) before spawning the editor so the tty
  is fully restored first
- Editor stdin/stdout/stderr explicitly redirected to /dev/tty so ncurses-
  based editors (nano, vim) work correctly when atuin's fds have been
  redirected by the shell integration's command-substitution fd-swap
- Out-of-bounds EditAccept (empty results list) returns the current input
  rather than panicking

wolf commented:
I've brought everything up-to-date; overhauled the implementation; tested by hand; mentioned in the CLI channel on discord. This work is ready for others to critique, try, review. I hope I've made something here useful to others.
